File: Backend - SpringBoot/src/main/java/com/estupiso/model/Actor.java
package com.estupiso.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;
import org.hibernate.validator.constraints.URL;

@Data
@Entity
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
public abstract class Actor extends DomainEntity {

    @NotBlank
    private String nombre;

    @NotBlank
    private String usuario;

    @NotBlank
    private String contraseña;

    @Email
    @NotBlank
    private String email;

    @URL
    private String fotoPerfil;

    private Roles rol;

    public Actor() {
    }
}

File: Frontend - Angular/src/app/model/Actor.ts
export enum Roles {
    ADMIN = 'ADMIN',
    ANUNCIANTE = 'ANUNCIANTE',
    ESTUDIANTE = 'ESTUDIANTE'
}


export abstract class Actor {
    id!: number;
    nombre: string;
    usuario: string;
    contraseña: string;
    email: string;
    fotoPerfil: string;
    rol: Roles;

    constructor(nombre: string, usuario: string, contraseña: string, email: string, fotoPerfil: string, rol: Roles) {
        this.nombre = nombre;
        this.usuario = usuario;
        this.contraseña = contraseña;
        this.email = email;
        this.fotoPerfil = fotoPerfil;
        this.rol = rol;
    }
}
